pgx-pg-sys/src/submodules/errcodes.rs

```diff
@@ -0,0 +1,1291 @@
+use std::fmt::{Display, Formatter};
+
+/// This list of SQL Error Codes is taken directly from Postgres 12's generated "utils/errcodes.h"
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub enum PgSqlErrorCode {
```

Comment:
Some musing to think about later that occurred to me when reading this:

I do wonder about having this be an enum rather than a `#[repr(transparent)] struct PgSqlErrorCode(c_int)`, along with a bunch of associated constants (you could still `match` on that so long as we use `#[derive]` to get PartialEq and friends).

That would let us avoid needing to translate all unknown values to `ERRCODE_INTERNAL_ERROR`.  would also potentially let extensions add their own error codes, but I don't know if that's a thing that makes sense at all (given that we defined `RustPanic` it *feels* like it might make sense, but I don't really know.

But yeah, I could easily be wrong, and even if I'm not -- we probably shouldn't do that in the same PR anyway :p.